src/v/storage/index_state.cc

```diff
@@ -583,17 +716,19 @@ index_state::find_below_size_bytes(size_t distance) {
     if (empty()) {
         return std::nullopt;
     }
-    auto it = std::upper_bound(
-      std::begin(position_index), std::end(position_index), distance);
+    auto it = index.position_upper_bound(distance);
+    if (it == std::nullopt) {
+        return std::nullopt;
+    }
 
-    if (it != position_index.begin()) {
-        it = std::prev(it);
```

Comment:
it looks to me like before this change if upper_bound returned end() that `it` would be set to prev(it) and fall through but now above if `it` is end() then we return nullopt. can you comment on how these are equivalent or how it is now correct?

Reply:
good catch, fixed